core/src/sigverify.rs

```diff
@@ -124,23 +121,18 @@ pub fn ed25519_verify(batches: &[SharedPackets]) -> Vec<Vec<u8>> {
 }
 
 pub fn get_packet_offsets(packet: &Packet, current_offset: u32) -> (u32, u32, u32, u32) {
-    // Read in the size of signatures array
-    let start_offset = TX_OFFSET + size_of::<u64>();
-    let mut rd = Cursor::new(&packet.data[start_offset..]);
-    let sig_len = decode_len(&mut rd).unwrap();
-    let sig_size = rd.position() as usize;
-    let msg_start_offset = start_offset + sig_size + sig_len * size_of::<Signature>();
-    let mut rd = Cursor::new(&packet.data[msg_start_offset..]);
-    let _ = decode_len(&mut rd).unwrap();
-    let pubkey_size = rd.position() as usize;
-    let pubkey_offset = current_offset as usize + msg_start_offset + pubkey_size;
-
-    let sig_start = start_offset + current_offset as usize + sig_size;
+    let (sig_len, sig_size) = decode_len(&packet.data);
+    let msg_start_offset = current_offset as usize + sig_size + sig_len * size_of::<Signature>();
+
+    let (_pubkey_len, pubkey_size) = decode_len(&packet.data[msg_start_offset..]);
+    let pubkey_offset = msg_start_offset + pubkey_size;
+
+    let sig_start = current_offset as usize + sig_size;
 
     (
         sig_len as u32,
         sig_start as u32,
-        current_offset + msg_start_offset as u32,
+        msg_start_offset as u32,
```

Comment:
@sakridge, there doesn't seem to be any test that checks this value. I can add `current_offset` or not right here, and all tests pass.  But upon closer inspection, I don't think there was a bug in the original code. And I don't *think* I added one.

Reply:
We should add a test that a vector with multiple different transactions works.

Reply:
Yeah, one with multiple signatures would make it less fragile too.

Reply:
ok, are you going to add back the add?

Reply:
I don't think I need to. I'm expanding the original test suite in an upcoming PR to be sure.

Reply:
yep, all good

Reply:
test_verify_seventy_one with --features=cuda enabled catches this. The CPU path doesn't use the current_offset (always passes 0), but the GPU path does.

Reply:
Did you see the test I added and rebased this PR on? Also, did you see how `current_offset` is now used a few lines prior? https://github.com/solana-labs/solana/pull/3545/files#diff-ad695894f5a727f4ce60e68987c4e35aR125